midterm/midterm/midterm_setup_Martin_Pecha.R

```diff
@@ -0,0 +1,350 @@
+  }
+  available_series <- data.frame(indicator=indicators,available=value)
+  return(available_series)
+}
+
+#more smooth vay of checking and creating whole vector and store it directly in data frame
+availability2 <- function(df,v){
+  value=indicators %in% names(df) 
+  available_series <- data.frame(indicator=indicators,available=value)
+  return(available_series)   
+}
+
+available_series <- availability(new_data,indicators)
+available_series2 <-availability2(new_data,indicators)
+
+rm(available_series2) #getting rid of one dataframe to reduce memory
+
+#1.3)
+# Create a slice of the dataframe for those indicators for which column available is TRUE. 
+# Call the result df_slice
+# You can either do that by creating a function or without creating a function
+
+df_slice <- available_series %>%
+  filter(available_series$available,TRUE) %>%
+  pull(1) #pulling only names of the columns to get a nice vector
+
+df_slice <- as.vector(df_slice) #transforming factors into a vector with characters
+
+#using dplyr
+selected_data <- new_data %>%
+  select(df_slice) 
+
+#practising by using a function (however too complicated)
+select_data <- function(df,v){
+  value=data.frame(nrow=dim(df)[1])#generating random dataset with one column and the same number of rows as df
+  for (i in 1:length(names(df))){
+    if (names(df[i]) %in% v == TRUE){
+    value <- cbind(value,df[i])  
+    }
+    else{
+    }
+  }
+  return(value[,-1])#removing a random generated column
+}
+
+alternative <- select_data(new_data,df_slice)
+
+rm(alternative)#removing alternative for the transparency
+
+#############################################################################################
+# Task 2
+#############################################################################################
+# You're gonna continue working with the slice only. Feel free to rename it and to remove any unnecessary
+# variables
+
+
+#2.1) Name of the columns are not really informative. Rename columns using col_names vector
+col_names = c("household_id","family_id","individual_id",'region','employment_status',
+              'year','age','race','hispanic','self_employed','sex','family_income')
+
+names(selected_data) <- col_names
+
+# 2.2)
+# Create a function that has dataframe df as an input and returns you the number of households,
+# number of families and number of individuals within the dataframe df
+
+#approach using dplyr package
+fun2 <- function(df){
+n_family <- selected_data %>%
+  count(family_id)%>%
+  summarise(unique_family=n_distinct(family_id,na.rm=TRUE))
+
+n_household <- selected_data %>%
+  count(household_id)%>%
+  summarise(unique_households=n_distinct(household_id,na.rm=TRUE))
+
+n_individual <- selected_data %>%
+  count(individual_id)%>%
+  summarise(unique_individuals=n_distinct(individual_id,na.rm=TRUE))
+
+return(c(n_household,n_family,n_individual))
+}
+
+
+fun2(selected_data)
+
+#There are 37451 housholds, 12 families and 15 individuals. 
+
```

Comment:
Unfortunately, there are more families and individuals
If you run  
df_slice %>%
  group_by(household_id, family_id) %>%
  count(individual_id) 
you  will see slightly different picture. Effectively, the only problem is that one particular family may be there for a few years. But you have aggregated family_id's from all the families

+ same comment regarding unique guys